File: budget-app-be/src/dto/filtering/query.ts
import {IComparableItem} from "./interfaces/comparable-item.interface";
import {FilterTypeEnum} from "./filter-type.enum";

export class Query implements IComparableItem {
  public fieldName: string = "";
  public fieldValue: string = "";
  public comparisonType: FilterTypeEnum = FilterTypeEnum.none;
}

File: Mettle.Backend/Mettle.Dms.Backend.Dto/Filtering/Query.cs
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;
using BudgetApp.Backend.Dto.Interfaces;

namespace BudgetApp.Backend.Dto.Filtering
{
    public class Query : IComparableItem
    {
        [JsonPropertyName("comparisonType")]
        public FilterType? ComparisonType { get; set; }

        [JsonPropertyName("typeDiscriminator")]
        public string TypeDiscriminator => nameof(Query);
        [JsonPropertyName("fieldName")]
        public string FieldName { get; set; } = "";
        [JsonPropertyName("fieldValue")]
        public List<string> FieldValue { get; set; } = new List<string>();
    }
}